be/src/Project/Services/DownloadCenter/DownloadCenter.Api/Services/DownloadCenter/DownloadCenterService.cs

```diff
@@ -0,0 +1,350 @@
+
+            var pagedResults = downloadCenterCategories.PagedIndex(new Pagination(downloadCenterCategories.Count(), model.ItemsPerPage), model.PageIndex);
+
+            var pagedDownloadCenterServiceModel = new PagedResults<IEnumerable<DownloadCenterItemServiceModel>>(pagedResults.Total, pagedResults.PageSize);
+
+            var downloadCenterItems = new List<DownloadCenterItemServiceModel>();
+
+            foreach (var downloadCenterItem in pagedResults.Data.OrEmptyIfNull().ToList())
+            {
+                var item = new DownloadCenterItemServiceModel
+                {
+                    Id = downloadCenterItem.Id,
+                    LastModifiedDate = downloadCenterItem.LastModifiedDate,
+                    CreatedDate = downloadCenterItem.CreatedDate
+                };
+
+                var categoryTranslation = this.context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == downloadCenterItem.Id && x.IsActive && x.Language == model.Language);
+
+                if (categoryTranslation is null)
+                {
+                    categoryTranslation = this.context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == downloadCenterItem.Id && x.IsActive);
+                }
+
+                item.Name = categoryTranslation?.Name;
+
+                var categories = this.context.Categories.Where(x => x.ParentCategoryId == downloadCenterItem.Id && x.IsActive && x.IsVisible);
+
+                var downloadCategories = new List<DownloadCenterSubcategoryServiceModel>();
+
+                foreach (var category in categories.OrEmptyIfNull().ToList())
+                {
+                    var categoryItem = new DownloadCenterSubcategoryServiceModel
+                    {
+                        Id = category.Id
+                    };
+
+                    var categoryItemTranslation = this.context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == category.Id && x.IsActive && x.Language == model.Language);
+
+                    if (categoryItemTranslation is null)
+                    {
+                        categoryItemTranslation = this.context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == category.Id && x.IsActive);
+                    }
+
+                    categoryItem.Name = categoryItemTranslation?.Name;
+
+                    downloadCategories.Add(categoryItem);
+                }
+
+                item.Categories = downloadCategories;
+
+                downloadCenterItems.Add(item);
+            }
+
+            pagedDownloadCenterServiceModel.Data = downloadCenterItems;
+
+            return pagedDownloadCenterServiceModel;
+        }
+
+        public async Task<DownloadCenterCategoryServiceModel> GetDownloadCenterCategoryAsync(GetDownloadCenterFilesCategoryServiceModel model)
+        {
+            var downloadCenterCategory = await this.context.Categories.FirstOrDefaultAsync(x => x.Id == model.Id && x.IsActive && x.IsVisible);
```

Comment:
**suggestion:** it's a very complicated method. It should be simplified (maybe split between multiple ones).

Reply:
@dawiddworak88  

In my opinion that this solution will not be bad. It works and gives the expected result. So why break it down into separate endpoints

Could this delay the ssr loading?